File: trees_and_graphs/find_if_path_exists_in_graph.java
class Solution
{
    public boolean validPath(int n, int[][] edges, int source, int destination)
    {
        Map<Integer, List<Integer>> graph = new HashMap<>();
        boolean[] seen = new boolean[n];

        for (int[] edge : edges)
        {
            int a = edge[0];
            int b = edge[1];
            graph.computeIfAbsent(a, val -> new ArrayList<Integer>()).add(b);
            graph.computeIfAbsent(b, val -> new ArrayList<Integer>()).add(a);
        }

        return dfsRecursive(graph, seen, source, destination);
    }

    private boolean dfsRecursive(Map<Integer, List<Integer>> graph,
                                 boolean[]                   seen,
                                 int                         currNode,
                                 int                         destination)
    {
        if (currNode == destination)
        {
            return true;
        }

        if (!seen[currNode])
        {
            seen[currNode] = true;
            for (int nextNode : graph.get(currNode))
            {
                if (dfsRecursive(graph, seen, nextNode, destination))
                {
                    return true;
                }
            }
        }

        return false;
    }
}

File: trees_and_graphs/find_if_path_exists_in_graph.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @param {number} source
 * @param {number} destination
 * @return {boolean}
 */
var validPath = function(n, edges, source, destination)
{
    let graph = new Map();
    for (let i = 0; i < n; i++)
    {
        graph.set(i, []);
    }

    for (const [a, b] of edges)
    {
        graph.get(a).push(b);
        graph.get(b).push(a);
    }

    let seen = new Array(n).fill(false);

    const dfsRecursive = currNode => {
        if (currNode == destination)
        {
            return true;
        }

        if (!seen[currNode])
        {
            seen[currNode] = true;
            for (const nextNode of graph.get(currNode))
            {
                if (dfsRecursive(nextNode))
                {
                    return true;
                }
            }
        }

        return false;
    }

    return dfsRecursive(source);
};